File: merge_css.js
const fs = require('fs');
const path = require('path');

// Use explicit absolute path to avoid CWD confusion
const dir = 'c:\\Users\\riote\\Downloads\\as_dance_full_project\\frontend\\src\\ui';
const stylesPath = path.join(dir, 'styles.css');
const headerPath = path.join(dir, 'header_new.css');

try {
    console.log('Reading ' + stylesPath);
    const stylesContent = fs.readFileSync(stylesPath, 'utf8');
    const stylesLines = stylesContent.split(/\r?\n/);
    console.log('Original lines: ' + stylesLines.length);

    console.log('Reading ' + headerPath);
    const headerContent = fs.readFileSync(headerPath, 'utf8');

    // Slice from 1600 (start of legacy)
    const tailLines = stylesLines.slice(1600);
    console.log('Tail lines: ' + tailLines.length);

    // Join with newline in case
    const tailContent = tailLines.join('\n');
    const finalContent = headerContent + "\n" + tailContent;

    console.log('Overwriting styles.css...');
    fs.writeFileSync(stylesPath, finalContent, 'utf8');
    console.log('SUCCESS: styles.css updated.');
} catch (e) {
    console.error('ERROR:', e);
    process.exit(1);
}

File: merge_css.py

import os

def merge_css():
    base_dir = r"c:\Users\riote\Downloads\as_dance_full_project\frontend\src\ui"
    styles_path = os.path.join(base_dir, "styles.css")
    header_path = os.path.join(base_dir, "header_new.css")
    
    print(f"Reading {styles_path}...")
    with open(styles_path, "r", encoding="utf-8") as f:
        lines = f.readlines()
        
    # Skip first 1600 lines (Old Header + Hero + Navbar)
    # Line 1600 in the original file was empty, 1599 was }
    # So we want to keep line 1601 onwards.
    # index 1600 is the 1601st line.
    legacy_lines = lines[1600:]
    
    print(f"Reading {header_path}...")
    with open(header_path, "r", encoding="utf-8") as f:
        header_lines = f.readlines()
        
    new_content = header_lines + legacy_lines
    
    print(f"Writing {len(new_content)} lines to {styles_path}...")
    with open(styles_path, "w", encoding="utf-8") as f:
        f.writelines(new_content)
        
    print("Merge complete.")

if __name__ == "__main__":
    merge_css()
